business/suggestionService.js

```diff
@@ -19,12 +19,30 @@ class SuggestionService {
   async get(coordinates) {
     const definitions = await this._getRelatedDefinitions(coordinates)
     if (!definitions) return null
+    let hasSuggested = false
     const suggestion = { coordinates }
-    this._collectSuggestionsForField(definitions, suggestion, 'licensed.declared', (definition, field) =>
-      isDeclaredLicense(get(definition, field))
-    )
-    this._collectSuggestionsForField(definitions, suggestion, 'described.sourceLocation')
-    return suggestion
+    if (Array.isArray(definitions.related) && definitions.related.length > 0) {
+      hasSuggested = this._collectSuggestionsForField(definitions, suggestion, 'licensed.declared', (definition, field) =>
+        isDeclaredLicense(get(definition, field))
+      )
+      hasSuggested |= this._collectSuggestionsForField(definitions, suggestion, 'described.sourceLocation')
+    }
+
+    if (!isDeclaredLicense(get(definitions.original, 'licensed.declared'))) {
+      let discoveredExpressions = get(definitions.original, 'licensed.facets.core.discovered.expressions')
+      if (Array.isArray(discoveredExpressions)) {
+        let appendDeclared = discoveredExpressions
+          .map(exp => { return { exp, cutAt: exp.indexOf('-') } })
+          .filter(marked => marked.cutAt > 0)
+          .map(marked => { return { value: marked.exp.substr(0, marked.cutAt), version: marked.exp.substr(marked.cutAt + 1) } })
```

Comment:
the 'version' here is not meant to be the license version.

it is meant to signify where the data was sourced from.
We should probably put  the original`coordinates.revision` in this property and leave the license expression in-tact

does that make sense?

Reply:
Oh, that was my bad. Yes I understand. Let me take care of it.

Reply:
Incorporated at https://github.com/clearlydefined/service/pull/507/commits/debdca3170bae574c80682f769097da9c47dfe4b